### PR TITLE
FEI-4022: implement useTimeout hook

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
This hook follows similar patterns to ActionScheduler.  I had to tweak things a little be since we're dropping the need for the scheduler itself with this hook.

I've also added lint rules for hooks in this PR.  I'll create a separate PR with that change and rebase this one on top of it.

Issue: FEI-4022

## Test plan:
- yarn jest use-timeout.test.js
- yarn start:storybook
- load http://localhost:6061/?path=/docs/timing-usetimeout
- see that the timeout starts immediately with the first example
- see that the timeout doesn't start until you click 'set' in the seconde example
- play around with setting and clearing the timeouts

### CODE REVIEW

File: packages/wonder-blocks-timing/src/hooks/use-timeout.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                window.clearTimeout(timeout);
+                if (!mountedRef.current) {
+                    clear();
+                }
+            };
+        }
+    }, [clear, isActive, timeoutMs]);
+
+    return {
+        get isSet() {
```
Comment (jeresig) [inline packages/wonder-blocks-timing/src/hooks/use-timeout.js:67]: is `isSet` clearer than `isActive`? Is there a harm in just doing: `isSet: isActive`?

**Review by jeresig — APPROVED** [review]
Looks good to me - thanks for the clear docs and tests!

File: packages/wonder-blocks-timing/src/hooks/use-timeout.js (hunk 1 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-timing/src/hooks/use-timeout.js:67]: `isSet` conforms to the `ITimeout` interface.  I going to rename `isActive` to `isSet`.  If we get feedback from devs that this name is confusing, we should rename it `ITimeout`.

I'm going to drop the getter method and use `isSet: isActive` instead.  The tests pass just fine after that change.  I'm also adding an additional test case to make sure that `isSet` is false after the timer expires.

File: package.json [hunk 2]
```diff
@@ -88,7 +89,7 @@
     "react-docgen": "^5.0.0",
     "react-router-enzyme-context": "^1.2.0",
     "react-styleguidist": "^11.0.5",
-    "react-test-renderer": "^16.4.1",
+    "react-test-renderer": "^16.9.0",
```
Comment (kevinbarabash) [inline package.json:92]: This was needed by `@testing-library/react-hooks`.

**Review by jandrade — APPROVED** [review]
LGTM! Just a minor suggestion for code formatting but the implementation and tests look great!.

File: packages/wonder-blocks-timing/src/hooks/use-timeout.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        },
+    }}
+/>
+
+# `useTimeout`
+
+`useTimeout` is a hook that provides a convenient API for setting and clearing
+a timeout. It is defined as follows:
+
+```
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-timing/src/hooks/use-timeout.stories.mdx]: I ended up going with `jsx` since it does a better job of highlighting things.

**Pull request merged**